### ISSUE TITLE
Generated override is not followed in gitattributes

### ISSUE THREAD

**Issue by Nixinova** [Nixinova/LinguistJS#26]
Having gitattributes content such as
```gitattributes
package-lock.json -linguist-generated
```
does not actually lead to the file being included in the output.

**Comment by Nixinova** [comment]
This occurs because default gitignored content is ignored when the list of files is being made so there is nothing to un-filter out when parsing gitattributes.

```ts
const data = walk({ init: true, commonRoot, folderRoots: resolvedInput, folders: resolvedInput, gitignores, regexIgnores });
files = data.files;
folders = data.folders;
```

**Comment by Nixinova** [comment]
Being worked on at the moment; requires quite a bit of a rewrite. Tests do not pass at this stage.

**Comment by Nixinova** [comment]
To be fixed in 2.7
